---
title: "Analysis CLL Proteomics - Final data Dimension Reduction"
author: "Sophie Rabe"
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_depth: 3
    toc_float: yes
    df_print: paged
  word_document:
    toc: yes
    toc_depth: '4'
  pdf_document:
    toc: yes
    toc_depth: '4'
editor_options:
  chunk_output_type: console
---
Analysis date: `r Sys.Date()`

# Setup
## Load libraries
```{r load packages, message=FALSE, warning=FALSE}
library(plyr)
library(gtools)
library(openxlsx)
library(pheatmap)
library(reshape2)
library(progress)
library(Matrix)
library(Hmisc)
library(lemon)
library(ggpubr)
library(effsize)
library(ggbeeswarm)
library(ggfortify)
library(ggpmisc)
library(ggrepel)
library(readxl)
library(DESeq2)
library(TOSTER)
library(tidyverse)
library(vsn)
library(fdrtool)
library(limma)
library(apeglm)
library(IHW)
library(Rtsne)
library(biomartr)
library(biomaRt)
library(MultiAssayExperiment)
library(PMA)
library(gplots)
library(RColorBrewer)
library(grid)
library(ConsensusClusterPlus)
library(survival)
library(survminer)
library(cowplot)
library(viridis)
```


## Load data
```{r load data, message=FALSE}
source("/Volumes/sd17b003/Sophie/Analysis/Screen_analysis/Figure_layouts.R")
load("/Volumes/sd17b003/Sophie/Analysis/CLL_Proteomics/CLL_Proteomics_final/Proteomics_Git/Robjects/CLL_Proteomics_Setup.RData")
load("/Volumes/sd17b003/Sophie/Analysis/CLL_Proteomics/CLL_Proteomics_final/Proteomics_Git/Robjects/CLL_Proteomics_ConsensusClustering.RData")
load("/Volumes/sd17b003/Sophie/Analysis/CLL_Proteomics/R_objects/GOterm_BCR.RData")
load("/Volumes/sd17b003/Sophie/Analysis/CLL_Proteomics/R_objects/GOterm_Spliceosome.RData")
```

```{r Add CCP group}
colData(multiomics_MAE)$PG <- as.factor(CCP_group5[rownames(colData(multiomics_MAE))])
colData(multiomics_MAE)$CCP6_RNA <- as.factor(CCP_group6_RNA[rownames(colData(multiomics_MAE))])
```

# Analysis
## Dimension reduction plots
### tsne (Barnes-Hut-SNE)
#### t-SNE colored by genetic alterations and groups
```{r tsne, cache=TRUE, dependson="consensus clustering" , dev=c("png","pdf"), message=FALSE, fig.height=5}
set.seed(10)
rtsne_out <- Rtsne( t(assay(multiomics_MAE[prot_few_nas , ,"proteomics"])), perplexity = 10 )
rtsne_out_df <- rtsne_out$Y %>% as.data.frame() %>% as_tibble()
rtsne_out_df$pat_ID <- colnames(assay(multiomics_MAE[prot_few_nas , ,"proteomics"]))

rtsne_out_df <- left_join(rtsne_out_df, 
                          wideFormat(multiomics_MAE[, ,c("SNPs","chrom_abber", "health_record_bin") ], colDataCols = c("gender", "treatment_status", "doehner_groups", "PG")) %>% as_tibble(), 
                          by=c("pat_ID"="primary") )

rtsne_out_df <- mutate_at(rtsne_out_df, colnames(rtsne_out_df %>% dplyr::select(SNPs_ATM:health_record_bin_treated)), as.logical)
rtsne_out_df <- rtsne_out_df %>% replace(is.na(.), "unknown")

message("t-SNE colored by DÃ¶hner groups")
ggplot(rtsne_out_df, aes(V1, V2)) +
  geom_point(aes(color=doehner_groups), size=3) +
  pp_sra +
  theme(aspect.ratio=1, plot.title = element_text(size = 30))

message("t-SNE colored by consensus cluster groups")
tsne_CCP_P_plot <-  ggplot(rtsne_out_df, aes(V1, V2)) +
  geom_point(aes(fill=PG), color="grey", shape=21) +
  scale_fill_manual(values = colors_CCP)+
  pp_sra +
  guides(color=guide_legend(title="PG")) 
tsne_CCP_P_plot+  theme(aspect.ratio=1, plot.title = element_text(size = 30))

message("t-SNE colored by combination of IGHV status and trisomy12")
tsne_IGHVtris_plot <- ggplot(rtsne_out_df %>% filter(health_record_bin_IGHV_mutated!="unknown", chrom_abber_trisomy12!="unknown"), aes(V1, V2)) +
  geom_point(aes(fill=interaction(health_record_bin_IGHV_mutated, chrom_abber_trisomy12)), color="grey", shape=21) +
  scale_fill_manual(values = colors_CCP[c(1,3,4,2)] )+
  pp_sra_noguides 
tsne_IGHVtris_plot +  theme(aspect.ratio=1, plot.title = element_text(size = 30)) #+

message("t-SNE colored by IGHV status")
ggplot(rtsne_out_df, aes(V1, V2)) +
          geom_point(aes_string(color="health_record_bin_IGHV_mutated"), size=3) +
    scale_color_manual(values = c( "#0571b0", "#ca0020", "grey"))  + 
      pp_sra +
      theme(aspect.ratio=1, plot.title = element_text(size = 30))+
  guides(color=guide_legend(title="IGHV_mutated"))

message("t-SNE colored by trisomy12")
ggplot(rtsne_out_df, aes(V1, V2)) +
          geom_point(aes_string(color="chrom_abber_trisomy12"), size=3) +
    scale_color_manual(values = c( "#0571b0", "#ca0020", "grey"))  + 
      pp_sra +
      theme(aspect.ratio=1, plot.title = element_text(size = 30))+
  guides(color=guide_legend(title="trisomy12"))

sapply(colnames(rtsne_out_df)[c(4,5,8:29)], function(var){
  print(ggplot(rtsne_out_df, aes(V1, V2)) +
          geom_point(aes_string(color=var), size=3) +
    scale_color_manual(values = c( "#0571b0", "#ca0020", "grey"))  + 
          ggtitle(gsub("SNPs_", "", gsub("chrom_abber_", "", gsub( "health_record_bin_", "", var) ))) +
      pp_sra_noguides +
      theme(aspect.ratio=1, plot.title = element_text(size = 30)) )
})

message("There is are trisomy12 negative patients which clusters with all of the other trisomy12 patients. Does they have a subclonal mutations?")
metadata(multiomics_MAE)$fish_df_clonsizes["trisomy12" ,] %>% 
  dplyr::select(rtsne_out_df %>% filter(chrom_abber_trisomy12==FALSE) %>% arrange(desc(V1)) %>% slice(1:2) %>% .$pat_ID)
```

## BCR signaling
### Different kinds of plots
```{r tsne BCR genes, cache=TRUE, dependson=c("tsne", "consensus clustering"),  dev=c("png","pdf"), fig.height=5}
BCR_genes_mean <- assay(multiomics_MAE[BCR_genes$symbol, ,"proteomics"]) %>% colMeans(na.rm = TRUE) %>% enframe()

BCR_CCP_P_plot <- left_join(BCR_genes_mean, 
          enframe(multiomics_MAE$PG, value = "PG"), 
          by=c("name")) %>%
  filter(!is.na(PG)) %>%
  ggplot(aes( PG, value ))+
  geom_boxplot(aes(fill=PG)) + geom_beeswarm() + 
  ggtitle("BCR protein abundance") + 
  scale_fill_manual(values=colors_CCP) +
  pp_sra +
  ylab("Mean abundance of BCR proteins")+
  #theme(legend.position = "bottom") +
  stat_compare_means(label = "p.signif", method = "t.test",
                     ref.group = ".all.", label.y = 0.2, hide.ns = TRUE) +
  guides(fill=guide_legend(title="PG"))
BCR_CCP_P_plot + theme(aspect.ratio=1) + 
  stat_compare_means(method = "anova", label.y = 0.22, hjust=0)
```

### Heatmaps
#### Selected BCR proteins
```{r}
# Smaller heatmap with only selected proteins
#sel_BCR <- c("ZAP70",  "IGHM", "CD79A", "CD79B", "LYN",  "SYK", "BLNK", "PLCG2",  "BTK",  "PTPN6", "NFATC2", 
#             "MAPK1", "MAP2K2", "NRAS", "MALT1", "BCL10", "PIK3CD","CD19",  "VAV3",
#             "AKT1", "IKBKB")

sel_BCR <- c("ZAP70",  "IGHM", "CD79A", "CD79B",  "SYK", "PLCG2",  "BTK",  "PTPN6", 
             "MAPK1",  "PIK3CD", "AKT1", "IKBKB")

tmp_BCR <- wideFormat(multiomics_MAE[sel_BCR, ,"proteomics"], colDataCols = c("PG", "IGHV" )) %>% 
  as_tibble()
tmp_BCR_mx <- tmp_BCR %>% dplyr::select(-primary, -PG, -IGHV) %>% as.matrix()
rownames(tmp_BCR_mx) <- tmp_BCR$primary
colnames(tmp_BCR_mx) <- gsub("proteomics_", "", colnames(tmp_BCR_mx))
tmp_BCR_anno <- tmp_BCR[, c("PG", "IGHV")] %>% as.data.frame()
rownames(tmp_BCR_anno) <- tmp_BCR$primary
tmp_BCR_anno$IGHV[tmp_BCR_anno$IGHV=="U"] <- "U-CLL"
tmp_BCR_anno$IGHV[tmp_BCR_anno$IGHV=="M"] <- "M-CLL"

ann_colors = list(
  PG=c("1"= colors_CCP[1], "2"= colors_CCP[2], "3"= colors_CCP[3], "4"= colors_CCP[4], "5"= colors_CCP[5], "6"= colors_CCP[6] ),
  IGHV=c("U-CLL"= "#0571b0", "M-CLL"= "#ca0020"))

tmp_BCR_anno$PG <- factor(tmp_BCR_anno$PG, levels = c(1:3,5,6,4))

breaks= seq(min(tmp_BCR_mx), max(tmp_BCR_mx), 0.1)^2
breaks= sort(c(-breaks, breaks))
breaks <- breaks[! (breaks < min(tmp_BCR_mx) | breaks > max(tmp_BCR_mx) )]

pat_order_hclust <- sapply(c(1:3,5,6,4), function(P){
  hc <- hclust(dist(tmp_BCR_mx[rownames(tmp_BCR_anno[tmp_BCR_anno$PG==P,]), ] ))
  hc$labels[hc$order]
}) %>% unlist

PG_BCR_proteins_pheatmap <- pheatmap(t(tmp_BCR_mx[pat_order_hclust, ]), 
         annotation_col = tmp_BCR_anno, annotation_colors = ann_colors, scale = "row", cluster_cols = FALSE, 
         color = inferno(length(breaks)), border_color = NA, 
         gaps_col = (which(!tmp_BCR_anno %>% rownames_to_column() %>% arrange(PG, IGHV) %>% .$PG %>% duplicated())-1)[-1],  
         breaks = breaks , cutree_rows = 4, show_colnames = F, treeheight_row = 0, fontsize_row = 5)
```

#### GO terms B-cell activation or BCR signaling combined
```{r BCR genes heatmap GO BC activation, cache=TRUE, dependson=c("consensus clustering"),  dev=c("png","pdf"), fig.height=5}
tmp_BCR_GO <- wideFormat(multiomics_MAE[unique(c(GO_BC_activation$hgnc_symbol, GO_BCR$hgnc_symbol)), ,"proteomics"], colDataCols = c("PG", "IGHV" )) %>% 
  as_tibble()
tmp_BCR_GO_mx <- tmp_BCR_GO %>% dplyr::select(-primary, -PG, -IGHV) %>% as.matrix()
rownames(tmp_BCR_GO_mx) <- tmp_BCR_GO$primary
colnames(tmp_BCR_GO_mx) <- gsub("proteomics_", "", colnames(tmp_BCR_GO_mx))
tmp_BCR_GO_anno <- tmp_BCR_GO[, c("PG", "IGHV")] %>% as.data.frame()
rownames(tmp_BCR_GO_anno) <- tmp_BCR_GO$primary
tmp_BCR_GO_anno$PG <- factor(tmp_BCR_GO_anno$PG, levels = c(1:3,5,6,4))

ann_colors = list(
  PG=c("1"= colors_CCP[1], "2"= colors_CCP[2], "3"= colors_CCP[3], "4"= colors_CCP[4], "5"= colors_CCP[5], "6"= colors_CCP[6] ))

breaks= seq(min(tmp_BCR_GO_mx), max(tmp_BCR_GO_mx), 0.1)^2
breaks= sort(c(-breaks, breaks))
breaks <- breaks[! (breaks < min(tmp_BCR_GO_mx) | breaks > max(tmp_BCR_GO_mx) )]

message("Heatmap of abundance BCR signaling proteins ordered according to PG")
PG_BCR_pheatmap <- pheatmap(t(tmp_BCR_GO_mx[(tmp_BCR_GO_anno %>% rownames_to_column() %>% arrange(PG, IGHV) %>% .$rowname), ]), 
         annotation_col = tmp_BCR_GO_anno, annotation_colors = ann_colors, scale = "row", cluster_cols = FALSE, 
         color = inferno(length(breaks)), border_color = NA, 
         gaps_col = (which(!tmp_BCR_GO_anno %>% rownames_to_column() %>% arrange(PG, IGHV) %>% .$PG %>% duplicated())-1)[-1],
         breaks = breaks , cutree_rows = 3, show_colnames = F, treeheight_row = 0, fontsize_row = 5)
```

## Spliceosome
#### KEGG
```{r tsne splice genes, cache=TRUE, dependson=c("tsne", "consensus clustering"),  dev=c("png","pdf"), fig.height=5}
splice_genes_mean <- assay(multiomics_MAE[splice_genes$symbol, ,"proteomics"]) %>% colMeans(na.rm = TRUE) %>% enframe()

splicing_CCP_P_plot <- left_join(splice_genes_mean, 
          enframe(multiomics_MAE$PG, value = "PG"), 
          by=c("name")) %>%
  filter(!is.na(PG)) %>%
  ggplot(aes( PG, value ))+
  geom_boxplot(aes(fill=PG)) + geom_beeswarm() + 
  ggtitle("Spliceosome protein abundance") + 
  pp_sra +
  ylab("Mean abundance of spliceosome proteins")+
  stat_compare_means(label = "p.signif", method = "t.test",
                     ref.group = ".all.", label.y = 0.2, hide.ns = TRUE)+
  guides(fill=guide_legend(title="PG")) +
  scale_fill_manual(values = colors_CCP)
splicing_CCP_P_plot +   theme(aspect.ratio=1) + 
    stat_compare_means(method = "anova", label.y = 0.22, hjust=0)
```

#### GO term Spliceosomal complex
```{r Splice genes heatmap GO, cache=TRUE, dependson=c("consensus clustering"),  dev=c("png","pdf"), fig.height=5}
tmp_Splice_GO <- wideFormat(multiomics_MAE[GO_SpliceosomalComplex$hgnc_symbol, ,"proteomics"], colDataCols = c("PG", "IGHV" )) %>% 
  as_tibble()
tmp_Splice_GO_mx <- tmp_Splice_GO %>% dplyr::select(-primary, -PG, -IGHV) %>% as.matrix()
rownames(tmp_Splice_GO_mx) <- tmp_Splice_GO$primary
colnames(tmp_Splice_GO_mx) <- gsub("proteomics_", "", colnames(tmp_Splice_GO_mx))
tmp_Splice_GO_anno <- tmp_Splice_GO[, c("PG", "IGHV")] %>% as.data.frame()
rownames(tmp_Splice_GO_anno) <- tmp_Splice_GO$primary
tmp_Splice_GO_anno$PG <- factor(tmp_Splice_GO_anno$PG,levels = c(1:3,5,6,4))

ann_colors = list(
  PG=c("1"= colors_CCP[1], "2"= colors_CCP[2], "3"= colors_CCP[3], "4"= colors_CCP[4], "5"= colors_CCP[5], "6"= colors_CCP[6] ))

breaks= seq(min(tmp_Splice_GO_mx), max(tmp_Splice_GO_mx), 0.1)^2
breaks= sort(c(-breaks, breaks))
breaks <- breaks[! (breaks < min(tmp_Splice_GO_mx) | breaks > max(tmp_Splice_GO_mx) )]

message("Heatmap of abundance spliceosome proteins ordered according to PG")
PG_GOSplice_pheatmap <- pheatmap(t(tmp_Splice_GO_mx[(tmp_Splice_GO_anno %>% rownames_to_column() %>% arrange(PG, IGHV) %>% .$rowname), ]), 
         annotation_col = tmp_Splice_GO_anno, annotation_colors = ann_colors, scale = "row", cluster_cols = FALSE, 
         color =  inferno(length(breaks)), border_color = NA, 
         gaps_col = (which(!tmp_Splice_GO_anno %>% rownames_to_column() %>% arrange(PG, IGHV) %>% .$PG %>% duplicated())-1)[-1],
         breaks = breaks , show_colnames = F , treeheight_row = 0, fontsize_row = 2)
```

#### Selected spliceosome proteins
```{r}
some_spliceprots <- c("SF3B1", "SNRPA", "PRPF6", "PRPF3", "SF3A1", "SNRPD2",  "SRSF4", "CDC5L", "PRPF19",
                      "CRNKL1", "PUF60", "PRPF8",
                      "SNRPB2")

tmp_Splice_GO <- wideFormat(multiomics_MAE[some_spliceprots, ,"proteomics"], colDataCols = c("PG", "IGHV" )) %>% 
  as_tibble()
tmp_Splice_GO_mx <- tmp_Splice_GO %>% dplyr::select(-primary, -PG, -IGHV) %>% as.matrix()
rownames(tmp_Splice_GO_mx) <- tmp_Splice_GO$primary
colnames(tmp_Splice_GO_mx) <- gsub("proteomics_", "", colnames(tmp_Splice_GO_mx))
tmp_Splice_GO_anno <- tmp_Splice_GO[, c("PG", "IGHV")] %>% as.data.frame()
rownames(tmp_Splice_GO_anno) <- tmp_Splice_GO$primary
tmp_Splice_GO_anno$PG <- factor(tmp_Splice_GO_anno$PG,levels = c(1:3,5,6,4))
SF3B1mut <- wideFormat(multiomics_MAE["SF3B1",,"SNPs"])$SNPs_SF3B1
names(SF3B1mut) <- wideFormat(multiomics_MAE["SF3B1",,"SNPs"])$primary
tmp_Splice_GO_anno$SF3B1 <- as.factor(SF3B1mut[rownames(tmp_Splice_GO_anno)])
tmp_Splice_GO_anno$IGHV[tmp_Splice_GO_anno$IGHV=="U"] <- "U-CLL"
tmp_Splice_GO_anno$IGHV[tmp_Splice_GO_anno$IGHV=="M"] <- "M-CLL"

ann_colors = list(
  PG=c("1"= colors_CCP[1], "2"= colors_CCP[2], "3"= colors_CCP[3], "4"= colors_CCP[4], "5"= colors_CCP[5], "6"= colors_CCP[6] ),
  SF3B1=c("1"="darkblue", "0"="gray80"),
  IGHV=c("U-CLL"= "#0571b0", "M-CLL"= "#ca0020"))

pat_order_hclust_splice <- sapply(c(1:3,5,6,4), function(P){
  hc <- hclust(dist(tmp_Splice_GO_mx[rownames(tmp_Splice_GO_anno[tmp_Splice_GO_anno$PG==P,]), ] ))
  hc$labels[hc$order]
}) %>% unlist

PG_splice_proteins_pheatmap <- pheatmap(t(tmp_Splice_GO_mx[pat_order_hclust_splice, ]), 
         annotation_col = tmp_Splice_GO_anno, annotation_colors = ann_colors, scale = "row", cluster_cols = FALSE, 
         color =  inferno(length(breaks)), border_color = NA, 
         gaps_col = (which(!tmp_Splice_GO_anno %>% rownames_to_column() %>% arrange(PG, IGHV) %>% .$PG %>% duplicated())-1)[-1],
         breaks = breaks , show_colnames = F , treeheight_row = 0, fontsize_row = 5)
```

## PCA
```{r PCA, cache=TRUE, dependson="consensus clustering", dev=c("png","pdf") , fig.height=5}
prot_pca <- prcomp(t( assay(multiomics_MAE[prot_few_nas , ,"proteomics"]) ))
prot_pca_x <- as_tibble(prot_pca$x[,1:10])
prot_pca_x$pat_ID <- colnames(assay(multiomics_MAE[prot_few_nas , ,"proteomics"]))


prot_pca_x <- left_join(prot_pca_x, 
                        wideFormat(multiomics_MAE[, ,c("SNPs","chrom_abber", "health_record_bin") ], 
                                   colDataCols = c("gender", "treatment_status", "PG")) %>% as_tibble() , 
                        by=c("pat_ID"="primary") )

prot_pca_x <- prot_pca_x %>% replace(is.na(.), "unknown")

ggplot(prot_pca_x, aes(PC1, PC2)) +
  geom_point(aes(color=health_record_bin_IGHV_mutated, shape=chrom_abber_trisomy12)) + pp_sra +
  scale_shape_manual(values = c( 16,  1, 4)) + 
  scale_color_manual(values = c( "#0571b0","#ca0020", "grey")) + theme(aspect.ratio = 1)
ggplot(prot_pca_x, aes(PC1, PC3)) +geom_point(aes(color=health_record_bin_IGHV_mutated, shape=chrom_abber_trisomy12)) +
  scale_shape_manual(values = c( 16,  1, 4)) + 
  scale_color_manual(values = c( "#0571b0","#ca0020", "grey"))+ pp_sra+ theme(aspect.ratio = 1)
ggplot(prot_pca_x, aes(PC2, PC3)) +geom_point(aes(color=health_record_bin_IGHV_mutated, shape=chrom_abber_trisomy12)) +  
  scale_shape_manual(values = c( 16,  1, 4)) + 
  scale_color_manual(values = c( "#0571b0", "#ca0020", "grey")) + pp_sra+ theme(aspect.ratio = 1)

ggplot(prot_pca_x, aes(PC1, PC2)) +geom_point(aes(color=gender)) + pp_sra + scale_color_manual(values = c( "#0571b0","#ca0020", "grey")) + theme(aspect.ratio = 1)

PCA_CCP_1_2 <- ggplot(prot_pca_x, aes(PC1, PC2)) +
  geom_point(aes(fill=PG),shape=21, color="grey") + pp_sra  +
  guides(color=guide_legend(title="PG")) +
  scale_fill_manual(values = colors_CCP) 
PCA_CCP_1_2 +
  theme(aspect.ratio=1, plot.title = element_text(size = 30))

PCA_CCP_1_3 <- ggplot(prot_pca_x, aes(PC1, PC3)) +
  geom_point(aes(fill=PG),shape=21, color="grey") + pp_sra  +
  guides(color=guide_legend(title="PG")) +
  scale_fill_manual(values = colors_CCP) 
PCA_CCP_1_3 +
  theme(aspect.ratio=1, plot.title = element_text(size = 30))

PCA_CCP_2_3 <- ggplot(prot_pca_x, aes(PC2, PC3)) +
  geom_point(aes(fill=PG),shape=21, color="grey") + pp_sra  +
  guides(color=guide_legend(title="PG")) +
  scale_fill_manual(values = colors_CCP) 
PCA_CCP_2_3 +
  theme(aspect.ratio=1, plot.title = element_text(size = 30))

PCA_CCP_1_4 <- ggplot(prot_pca_x, aes(PC1, PC4)) +
  geom_point(aes(fill=PG),shape=21, color="grey") + pp_sra  +
  guides(color=guide_legend(title="PG")) +
  scale_fill_manual(values = colors_CCP) 
PCA_CCP_1_4 +
  theme(aspect.ratio=1, plot.title = element_text(size = 30))
```

## tsne RNA
```{r tsne RNA, cache=TRUE, dev=c("png","pdf"), message=FALSE, fig.height=5}
genes_no_nas <- multiomics_MAE[["RNAseq_norm"]] %>% is.na() %>% rowSums()
genes_no_nas <- genes_no_nas[ genes_no_nas == 0 ] %>% names()

set.seed(10)
rtsne_out_RNA <- Rtsne( t(assay(multiomics_MAE[genes_no_nas , ,"RNAseq_norm"])), perplexity = 10 )
rtsne_out_RNA_df <- rtsne_out_RNA$Y %>% as.data.frame() %>% as_tibble()
rtsne_out_RNA_df$pat_ID <- colnames(assay(multiomics_MAE[genes_no_nas , ,"RNAseq_norm"]))

rtsne_out_RNA_df <- left_join(rtsne_out_RNA_df, 
                          (wideFormat(multiomics_MAE[, ,c("SNPs","chrom_abber", "health_record_bin") ], colDataCols = c("gender", "treatment_status", "doehner_groups", "PG", "CCP6_RNA")) %>% as_tibble()), 
                          by=c("pat_ID"="primary") )

rtsne_out_RNA_df <- mutate_at(rtsne_out_RNA_df, colnames(rtsne_out_RNA_df %>% dplyr::select(SNPs_ATM:health_record_bin_treated)), as.logical)

rtsne_out_RNA_df <- rtsne_out_RNA_df %>% replace(is.na(.), "unknown")


ggplot(rtsne_out_RNA_df, aes(V1, V2)) +
  geom_point(aes(color=doehner_groups), size=3) +
  scale_color_manual(values = c("#377eb8","#e41a1c", "#984ea3", "#4daf4a", "#ff7f00", "grey"))+
  pp_sra +
  theme(aspect.ratio=1, plot.title = element_text(size = 30))

tsne_CCP_R_plot <- ggplot(rtsne_out_RNA_df, aes(V1, V2)) +
  geom_point(aes(fill=PG), color="grey", shape=21) +
  scale_fill_manual(values = colors_CCP)+
  pp_sra +
  guides(color=guide_legend(title="PG"))
tsne_CCP_R_plot + theme(aspect.ratio=1, plot.title = element_text(size = 30))

tsne_CCPRNA_R_plot <- ggplot(rtsne_out_RNA_df, aes(V1, V2)) +
  geom_point(aes(fill=CCP6_RNA), color="grey", shape=21) +
  scale_color_hue()+
  pp_sra +
  guides(color=guide_legend(title="CC_RNA"))
tsne_CCPRNA_R_plot+  theme(aspect.ratio=1, plot.title = element_text(size = 30))

sapply(colnames(rtsne_out_RNA_df)[c(4,5,9:30)], function(var){
  print(ggplot(rtsne_out_RNA_df, aes(V1, V2)) +
          geom_point(aes_string(color=var), size=3) +
    scale_color_manual(values = c("#92c5de", "#f4a582",  "grey"))  + 
          ggtitle(gsub("SNPs_", "", gsub("chrom_abber_", "", gsub( "health_record_bin_", "", var) ))) +
      pp_sra_noguides +
      theme(aspect.ratio=1, plot.title = element_text(size = 30)) )
})

```

# Session Info
```{r sessionInfo}
sessionInfo()
```
