---
title: "Analysis CLL Proteomics - Dimension Reduction"
author: "Sophie Rabe"
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_depth: 3
    toc_float: yes
    df_print: paged
  word_document:
    toc: yes
    toc_depth: '4'
  pdf_document:
    toc: yes
    toc_depth: '4'
editor_options:
  chunk_output_type: console
---
Analysis date: `r Sys.Date()`

# Setup
## Load libraries
```{r load packages, message=FALSE, warning=FALSE}
library(plyr)
library(gtools)
library(pheatmap)
library(Matrix)
library(Hmisc)
library(ggpubr)
library(ggbeeswarm)
library(DESeq2)
library(tidyverse)
library(vsn)
library(fdrtool)
library(limma)
library(Rtsne)
library(MultiAssayExperiment)
library(gplots)
library(viridis)
```

## Load data
```{r load data, message=FALSE}
source("data/Figure_layouts.R")
load("data/CLL_Proteomics_Setup.RData")
load("data/CLL_Proteomics_ConsensusClustering.RData")

load("data/GOterm_BCR.RData")
load("data/GOterm_Spliceosome.RData")
```

```{r Add CCP group}
new_colData <- left_join(colData(multiomics_MAE) %>% as_tibble(),
  enframe(CCP_group5, value = "PG", name = "patient_ID") ) 

new_colData <- left_join(new_colData,
  enframe(CCP_group6_RNA, value = "CCP6_RNA", name = "patient_ID") ) %>%
  as.data.frame() %>%
  mutate(rn = patient_ID) %>%
  column_to_rownames(., var = "rn")

colData(multiomics_MAE)$PG <- as.factor(new_colData$PG)
colData(multiomics_MAE)$CCP6_RNA <- as.factor(new_colData$CCP6_RNA)
```

# Analysis
## Dimension reduction plots
### tsne (Barnes-Hut-SNE)
#### t-SNE colored by genetic alterations and groups
```{r tsne,  message=FALSE, fig.height=5}
set.seed(10)
rtsne_out <- Rtsne( t(assay(multiomics_MAE[prot_few_nas , ,"proteomics"])), perplexity = 10 )
rtsne_out_df <- rtsne_out$Y %>% as.data.frame() %>% as_tibble()
rtsne_out_df$pat_ID <- colnames(assay(multiomics_MAE[prot_few_nas , ,"proteomics"]))

rtsne_out_df <- left_join(rtsne_out_df, 
                          wideFormat(multiomics_MAE[, ,c("SNPs","chrom_abber", "health_record_bin") ], colDataCols = c("gender", "treatment_status", "doehner_groups", "PG")) %>% as_tibble(), 
                          by=c("pat_ID"="primary") )

rtsne_out_df <- mutate_at(rtsne_out_df, colnames(rtsne_out_df %>% dplyr::select(SNPs_ATM:health_record_bin_treated)), as.logical)
#rtsne_out_df <- rtsne_out_df %>% replace(is.na(.), "unknown")

message("t-SNE colored by consensus cluster groups")
tsne_CCP_P_plot <-  ggplot(rtsne_out_df, aes(V1, V2)) +
  geom_point(aes(fill=PG), color="grey", shape=21) +
  scale_fill_manual(values = colors_CCP)+
  pp_sra +
  guides(color=guide_legend(title="PG")) 
tsne_CCP_P_plot+  theme(aspect.ratio=1, plot.title = element_text(size = 30))

message("t-SNE colored by combination of IGHV status and trisomy12")
tsne_IGHVtris_plot <- ggplot(rtsne_out_df %>% filter(health_record_bin_IGHV_mutated!="unknown", chrom_abber_trisomy12!="unknown"), aes(V1, V2)) +
  geom_point(aes(fill=interaction(health_record_bin_IGHV_mutated, chrom_abber_trisomy12)), color="grey", shape=21) +
  scale_fill_manual(values = colors_CCP[c(1,3,4,2)] )+
  pp_sra_noguides 
tsne_IGHVtris_plot +  theme(aspect.ratio=1, plot.title = element_text(size = 30)) #+

sapply(colnames(rtsne_out_df)[c(4,5,8:22)], function(var){
  print(var)
  print(ggplot(rtsne_out_df, aes(V1, V2)) +
          geom_point(aes_string(color=var), size=3) +
    scale_color_manual(values = c( "#0571b0", "#ca0020", "grey"))  + 
          ggtitle(gsub("SNPs_", "", gsub("chrom_abber_", "", gsub( "health_record_bin_", "", var) ))) +
      pp_sra_noguides +
      theme(aspect.ratio=1, plot.title = element_text(size = 30)) )
})
```

```{r}
message("There is one trisomy12 negative patient which clusters with all of the other trisomy12 patients. Does it have a subclonal mutations?")
metadata(multiomics_MAE)$fish_df_clonsizes["trisomy12" ,] %>% 
  dplyr::select(rtsne_out_df %>% filter(chrom_abber_trisomy12==FALSE) %>% arrange(desc(V1)) %>% slice(1:2) %>% .$pat_ID)
```


## BCR signaling
### Boxplots mean abundance
```{r tsne BCR genes, fig.height=5}
BCR_genes_mean <- assay(multiomics_MAE[BCR_genes, ,"proteomics"]) %>% colMeans(na.rm = TRUE) %>% enframe()

left_join(BCR_genes_mean, 
          wideFormat(multiomics_MAE[ c("IGHV_mutated", "trisomy12"), ,]) %>% as_tibble(), 
          by=c("name"="primary")) %>%
  mutate_at(.vars =c("chrom_abber_trisomy12", "health_record_bin_IGHV_mutated" ), as.logical ) %>%
  filter(!is.na(chrom_abber_trisomy12)) %>%
  ggplot(aes( chrom_abber_trisomy12, value ))+
  geom_boxplot(aes(fill=chrom_abber_trisomy12)) + geom_beeswarm() + 
  stat_compare_means(method = "t.test", label = "p.signif", comparisons = list(c("TRUE", "FALSE"))) + 
  ggtitle("BCR protein abundance") + 
  scale_fill_manual(values=c("#0571b0", "#ca0020")) +
  pp_sra_nox +
  ylab("Mean abundance of BCR proteins")+
  theme(aspect.ratio=2) +
  theme(legend.position = "bottom")

left_join(BCR_genes_mean, 
          wideFormat(multiomics_MAE[ c("IGHV_mutated", "trisomy12"), ,]) %>% as_tibble(), 
          by=c("name"="primary")) %>%
  mutate_at(.vars =c("chrom_abber_trisomy12", "health_record_bin_IGHV_mutated" ), as.logical ) %>%
  filter(!is.na(health_record_bin_IGHV_mutated)) %>%
  ggplot(aes( health_record_bin_IGHV_mutated, value ))+
  geom_boxplot(aes(fill=health_record_bin_IGHV_mutated)) + geom_beeswarm() + 
  stat_compare_means(method = "t.test", label = "p.signif", comparisons = list(c("TRUE", "FALSE"))) + 
  ggtitle("BCR protein abundance") + 
  scale_fill_manual(values=c("#0571b0", "#ca0020")) +
  pp_sra_nox +
  ylab("Mean abundance of BCR proteins")+
  theme(aspect.ratio=2) +
  theme(legend.position = "bottom")

BCR_CCP_P_plot <- left_join(BCR_genes_mean, 
          colData(multiomics_MAE) %>% as_tibble() %>% select(name=patient_ID, PG ),
          by=c("name")) %>%
  filter(!is.na(PG)) %>%
  ggplot(aes( PG, value ))+
  geom_boxplot(aes(fill=PG)) + geom_beeswarm() + 
  ggtitle("BCR protein abundance") + 
  scale_fill_manual(values=colors_CCP) +
  pp_sra +
  ylab("Mean abundance of BCR proteins")+
  #stat_compare_means(label = "p.signif", method = "t.test",
  #                   ref.group = ".all.", label.y = 0.2, hide.ns = TRUE) +
  guides(fill=guide_legend(title="PG"))
BCR_CCP_P_plot + theme(aspect.ratio=1) + 
  stat_compare_means(method = "anova", label.y = 0.22, hjust=0)
```

### Heatmaps
#### Selected BCR proteins
```{r}
sel_BCR <- c("ZAP70",  "IGHM", "CD79A", "CD79B",  "SYK", "PLCG2",  "BTK",  "PTPN6", 
             "MAPK1",  "PIK3CD", "AKT1", "IKBKB")

tmp_BCR <- wideFormat(multiomics_MAE[sel_BCR, ,"proteomics"], colDataCols = c("PG", "IGHV_mutated" )) %>% 
  as_tibble()
tmp_BCR_mx <- tmp_BCR %>% dplyr::select(-primary, -PG, -IGHV_mutated) %>% as.matrix()
rownames(tmp_BCR_mx) <- tmp_BCR$primary
colnames(tmp_BCR_mx) <- gsub("proteomics_", "", colnames(tmp_BCR_mx))
tmp_BCR_anno <- tmp_BCR[, c("PG", "IGHV_mutated")] %>% as.data.frame()
rownames(tmp_BCR_anno) <- tmp_BCR$primary
tmp_BCR_anno$IGHV[tmp_BCR_anno$IGHV_mutated==0] <- "U-CLL"
tmp_BCR_anno$IGHV[tmp_BCR_anno$IGHV_mutated==1] <- "M-CLL"
tmp_BCR_anno <- tmp_BCR_anno %>% select(-IGHV_mutated)

ann_colors = list(
  PG=c("1"= colors_CCP[1], "2"= colors_CCP[2], "3"= colors_CCP[3], "4"= colors_CCP[4], "5"= colors_CCP[5], "6"= colors_CCP[6] ),
  IGHV=c("U-CLL"= "#0571b0", "M-CLL"= "#ca0020"))

tmp_BCR_anno$PG <- factor(tmp_BCR_anno$PG, levels = c(5,1:4,6))

breaks= seq(min(tmp_BCR_mx), max(tmp_BCR_mx), 0.1)^2
breaks= sort(c(-breaks, breaks))
breaks <- breaks[! (breaks < min(tmp_BCR_mx) | breaks > max(tmp_BCR_mx) )]

pat_order_hclust <- sapply(c(5,1:4,6), function(P){
  hc <- hclust(dist(tmp_BCR_mx[rownames(tmp_BCR_anno[tmp_BCR_anno$PG==P,]), ] ))
  hc$labels[hc$order]
}) %>% unlist

PG_BCR_proteins_pheatmap <- pheatmap(t(tmp_BCR_mx[pat_order_hclust, ]), 
         annotation_col = tmp_BCR_anno, annotation_colors = ann_colors, scale = "row", cluster_cols = FALSE, 
         color = inferno(length(breaks)), border_color = NA, 
         gaps_col = (which(!tmp_BCR_anno %>% rownames_to_column() %>% arrange(PG, IGHV) %>% .$PG %>% duplicated())-1)[-1],  
         breaks = breaks , cutree_rows = 4, show_colnames = F, treeheight_row = 0, fontsize_row = 8)

PG_BCR_proteins_pheatmap_notscaled <- pheatmap(t(tmp_BCR_mx[pat_order_hclust, ]), 
         annotation_col = tmp_BCR_anno, annotation_colors = ann_colors, cluster_cols = FALSE, 
         color = inferno(length(breaks)), border_color = NA, 
         gaps_col = (which(!tmp_BCR_anno %>% rownames_to_column() %>% arrange(PG, IGHV) %>% .$PG %>% duplicated())-1)[-1],  
         cutree_rows = 4, show_colnames = F, treeheight_row = 0, fontsize_row = 8)
```

#### All KEGG BCR signaling proteins
```{r tsne BCR genes heatmap, fig.height=5}
tmp_BCR <- wideFormat(multiomics_MAE[BCR_genes, ,"proteomics"], colDataCols = c("PG", "IGHV_mutated" )) %>% 
  as_tibble()
tmp_BCR_mx <- tmp_BCR %>% dplyr::select(-primary, -PG, -IGHV_mutated) %>% as.matrix()
rownames(tmp_BCR_mx) <- tmp_BCR$primary
colnames(tmp_BCR_mx) <- gsub("proteomics_", "", colnames(tmp_BCR_mx))
tmp_BCR_anno <- tmp_BCR[, c("PG", "IGHV_mutated")] %>% as.data.frame()
tmp_BCR_anno$PG <- factor(tmp_BCR_anno$PG,levels = c(5,1:4,6)) #
rownames(tmp_BCR_anno) <- tmp_BCR$primary

ann_colors = list(
  PG=c("1"= colors_CCP[1], "2"= colors_CCP[2], "3"= colors_CCP[3], "4"= colors_CCP[4], "5"= colors_CCP[5], "6"= colors_CCP[6] ))

message("Heatmap of abundance BCR signaling proteins ordered according to PG")
pheatmap(tmp_BCR_mx[(tmp_BCR_anno %>% rownames_to_column() %>% arrange(PG, IGHV_mutated) %>% .$rowname), ], 
         annotation_row = tmp_BCR_anno, annotation_colors = ann_colors, scale = "column", cluster_rows = FALSE, color = inferno(10), border_color = NA, 
         gaps_row = (which(!tmp_BCR_anno %>% rownames_to_column() %>% arrange(PG, IGHV_mutated) %>% .$PG %>% duplicated())-1)[-1])
```

## Spliceosome
#### All KEGG proteins
```{r tsne splice genes, fig.height=5}
splice_genes_mean <- assay(multiomics_MAE[splice_genes, ,"proteomics"]) %>% colMeans(na.rm = TRUE) %>% enframe()

splicing_CCP_P_plot <- left_join(splice_genes_mean, 
          colData(multiomics_MAE) %>% as_tibble() %>% select(name=patient_ID, PG ), 
          by=c("name")) %>%
  filter(!is.na(PG)) %>%
  ggplot(aes( PG, value ))+
  geom_boxplot(aes(fill=PG)) + geom_beeswarm() + 
  ggtitle("Spliceosome protein abundance") + 
  pp_sra +
  ylab("Mean abundance of spliceosome proteins")+
  #stat_compare_means(label = "p.signif", method = "t.test",
  #                   ref.group = ".all.", label.y = 0.2, hide.ns = TRUE)+
  guides(fill=guide_legend(title="PG")) +
  scale_fill_manual(values = colors_CCP)
splicing_CCP_P_plot +   theme(aspect.ratio=1) + 
    stat_compare_means(method = "anova", label.y = 0.22, hjust=0)

tmp_splice <- wideFormat(multiomics_MAE[splice_genes, ,"proteomics"], colDataCols = c("PG", "IGHV_mutated")) %>% as_tibble()
tmp_splice_mx <- tmp_splice %>% dplyr::select(-primary, -PG, -IGHV_mutated) %>% as.matrix()
rownames(tmp_splice_mx) <- tmp_splice$primary
colnames(tmp_splice_mx) <- gsub("proteomics_", "", colnames(tmp_splice_mx))
tmp_splice_anno <- tmp_splice %>% dplyr::select(PG, IGHV_mutated) %>% as.data.frame()
rownames(tmp_splice_anno) <- tmp_splice$primary
tmp_splice_anno$PG <- factor(tmp_splice_anno$PG,levels = c(5,1:4,6))

message("Heatmap of abundance spliceosome proteins ordered according to PG")
pheatmap(tmp_splice_mx[(tmp_splice_anno %>% rownames_to_column() %>% arrange( PG, IGHV_mutated) %>% .$rowname), ], 
         annotation_row = tmp_splice_anno, annotation_colors = ann_colors, scale = "column", cluster_rows = FALSE, color = inferno(10), border_color = NA,
         gaps_row = (which(!tmp_splice_anno %>% rownames_to_column() %>% arrange(PG, IGHV_mutated) %>% .$PG %>% duplicated())-1)[-1])
```


#### Selected spliceosome proteins
```{r}
some_spliceprots <- c("SF3B1", "SNRPA", "PRPF6", "PRPF3", "SF3A1", "SNRPD2",  "SRSF4", "CDC5L", "PRPF19",
                      "CRNKL1", "PUF60", "PRPF8",
                      "SNRPB2")

tmp_Splice_GO <- wideFormat(multiomics_MAE[some_spliceprots, ,"proteomics"], colDataCols = c("PG", "IGHV_mutated" )) %>% 
  as_tibble()
tmp_Splice_GO_mx <- tmp_Splice_GO %>% dplyr::select(-primary, -PG, -IGHV_mutated) %>% as.matrix()
rownames(tmp_Splice_GO_mx) <- tmp_Splice_GO$primary
colnames(tmp_Splice_GO_mx) <- gsub("proteomics_", "", colnames(tmp_Splice_GO_mx))
tmp_Splice_GO_anno <- tmp_Splice_GO[, c("PG", "IGHV_mutated")] %>% as.data.frame()
rownames(tmp_Splice_GO_anno) <- tmp_Splice_GO$primary
tmp_Splice_GO_anno$PG <- factor(tmp_Splice_GO_anno$PG,levels = c(5,1:4,6))
SF3B1mut <- wideFormat(multiomics_MAE["SF3B1",,"SNPs"])$SNPs_SF3B1
names(SF3B1mut) <- wideFormat(multiomics_MAE["SF3B1",,"SNPs"])$primary
tmp_Splice_GO_anno$SF3B1 <- SF3B1mut[rownames(tmp_Splice_GO_anno)]
tmp_Splice_GO_anno$SF3B1[tmp_Splice_GO_anno$SF3B1 == 1] <- "mut"
tmp_Splice_GO_anno$SF3B1[tmp_Splice_GO_anno$SF3B1==0] <- "wt"
tmp_Splice_GO_anno$SF3B1 <- as.factor(tmp_Splice_GO_anno$SF3B1)
tmp_Splice_GO_anno$IGHV[tmp_Splice_GO_anno$IGHV_mutated==0] <- "U-CLL"
tmp_Splice_GO_anno$IGHV[tmp_Splice_GO_anno$IGHV_mutated==1] <- "M-CLL"
tmp_Splice_GO_anno <- tmp_Splice_GO_anno %>% select(-IGHV_mutated)

ann_colors = list(
  PG=c("1"= colors_CCP[1], "2"= colors_CCP[2], "3"= colors_CCP[3], "4"= colors_CCP[4], "5"= colors_CCP[5], "6"= colors_CCP[6] ),
  SF3B1=c("mut"="darkblue", "wt"="gray80"),
  IGHV=c("U-CLL"= "#0571b0", "M-CLL"= "#ca0020"))

pat_order_hclust_splice <- sapply(c(5,1:4,6), function(P){
  hc <- hclust(dist(tmp_Splice_GO_mx[rownames(tmp_Splice_GO_anno[tmp_Splice_GO_anno$PG==P,]), ] ))
  hc$labels[hc$order]
}) %>% unlist

PG_splice_proteins_pheatmap <- pheatmap(t(tmp_Splice_GO_mx[pat_order_hclust_splice, ]), 
         annotation_col = tmp_Splice_GO_anno, annotation_colors = ann_colors, scale = "row", cluster_cols = FALSE, 
         color =  inferno(length(breaks)), border_color = NA, 
         gaps_col = (which(!tmp_Splice_GO_anno %>% rownames_to_column() %>% arrange(PG, IGHV) %>% .$PG %>% duplicated())-1)[-1],
         breaks = breaks , show_colnames = F , treeheight_row = 0, fontsize_row = 8)

PG_splice_proteins_pheatmap_notscaled <- pheatmap(t(tmp_Splice_GO_mx[pat_order_hclust_splice, ]), 
         annotation_col = tmp_Splice_GO_anno, annotation_colors = ann_colors, cluster_cols = FALSE, 
         color =  inferno(length(breaks)), border_color = NA, 
         gaps_col = (which(!tmp_Splice_GO_anno %>% rownames_to_column() %>% arrange(PG, IGHV) %>% .$PG %>% duplicated())-1)[-1],
          show_colnames = F , treeheight_row = 0, fontsize_row = 8)
```


### Spliceosome abundance across groups with SF3B1 mutational status color coded
```{r}
left_join(splice_genes_mean, 
          wideFormat(multiomics_MAE["SF3B1", , "SNPs"], colDataCols = "PG") %>% as_tibble(), 
          by=c("name"="primary")) %>%
  filter(!is.na(PG)) %>%
  ggplot(aes( PG, value ))+
  geom_boxplot(aes(fill=PG)) + geom_beeswarm(aes(color=as.factor(SNPs_SF3B1))) + 
  stat_compare_means(method = "anova", label.y = 0.22, hjust=0) + 
  ggtitle("Spliceosome protein abundance") + 
  pp_sra_nox +
  ylab("Mean abundance of spliceosome proteins")+
  theme(aspect.ratio=1) +
  stat_compare_means(label = "p.signif", method = "t.test",
                     ref.group = ".all.", label.y = 0.2, hide.ns = TRUE)+
  guides(fill=guide_legend(title="PG"))+
  scale_fill_manual(values = colors_CCP) +
  scale_color_manual(values = c("grey", "red"))
```

## BCAA
#### KEGG Valine, leucine and isoleucine degradation
```{r BCAA boxplot, fig.height=5}
BCAA_genes_mean <- assay(multiomics_MAE[BCAA_genes, ,"proteomics"]) %>% colMeans(na.rm = TRUE) %>% enframe()

BCAA_CCP_P_plot <- left_join(BCAA_genes_mean, 
          colData(multiomics_MAE) %>% as_tibble() %>% select(name=patient_ID, PG ), 
          by=c("name")) %>%
  filter(!is.na(PG)) %>%
  ggplot(aes( PG, value ))+
  geom_boxplot(aes(fill=PG)) + geom_beeswarm() + 
  ggtitle("BCAA protein abundance") + 
  pp_sra +
  ylab("Mean abundance of BCAA proteins")+
  #stat_compare_means(label = "p.signif", method = "t.test",
  #                   ref.group = ".all.", label.y = 0.55, hide.ns = TRUE)+
  guides(fill=guide_legend(title="PG")) +
  scale_fill_manual(values = colors_CCP)
BCAA_CCP_P_plot +   theme(aspect.ratio=1) + 
    stat_compare_means(method = "anova", label.y = 0.57, hjust=0)
```

#### Heatmap Selected proteins
```{r}
some_BCAA <- c("ACAT1", "PCCA", "PCCB", "HADH", "HADHB", "HADHA", "HIBADH", "MCCC1", "MCCC2", "MLYCD")

tmp_BCAA <- wideFormat(multiomics_MAE[some_BCAA, ,"proteomics"], colDataCols = c("PG", "IGHV_mutated" )) %>% 
  as_tibble()
tmp_BCAA_mx <- tmp_BCAA %>% dplyr::select(-primary, -PG, -IGHV_mutated) %>% as.matrix()
rownames(tmp_BCAA_mx) <- tmp_BCAA$primary
colnames(tmp_BCAA_mx) <- gsub("proteomics_", "", colnames(tmp_BCAA_mx))
tmp_BCAA_anno <- tmp_BCAA[, c("PG", "IGHV_mutated")] %>% as.data.frame()
rownames(tmp_BCAA_anno) <- tmp_BCAA$primary
tmp_BCAA_anno$PG <- factor(tmp_BCAA_anno$PG,levels = c(5,1:4,6))
tmp_BCAA_anno$IGHV[tmp_BCAA_anno$IGHV_mutated==0] <- "U-CLL"
tmp_BCAA_anno$IGHV[tmp_BCAA_anno$IGHV_mutated==1] <- "M-CLL"
tmp_BCAA_anno <- tmp_BCAA_anno %>% select(-IGHV_mutated)

breaks= seq(min(tmp_BCAA_mx), max(tmp_BCAA_mx), 0.1)^2
breaks= sort(c(-breaks, breaks))
breaks <- breaks[! (breaks < min(tmp_BCAA_mx) | breaks > max(tmp_BCAA_mx) )]

ann_colors = list(
  PG=c("1"= colors_CCP[1], "2"= colors_CCP[2], "3"= colors_CCP[3], "4"= colors_CCP[4], "5"= colors_CCP[5], "6"= colors_CCP[6] ),
  IGHV=c("U-CLL"= "#0571b0", "M-CLL"= "#ca0020"))

pat_order_hclust_splice <- sapply(c(5,1:4,6), function(P){
  hc <- hclust(dist(tmp_BCAA_mx[rownames(tmp_BCAA_anno[tmp_BCAA_anno$PG==P,]), ] ))
  hc$labels[hc$order]
}) %>% unlist

PG_BCAA_proteins_pheatmap <- pheatmap(t(tmp_BCAA_mx[pat_order_hclust_splice, ]), 
         annotation_col = tmp_BCAA_anno, annotation_colors = ann_colors, scale = "row", cluster_cols = FALSE, 
         color =  inferno(length(breaks)), border_color = NA, 
         gaps_col = (which(!tmp_BCAA_anno %>% rownames_to_column() %>% arrange(PG, IGHV) %>% .$PG %>% duplicated())-1)[-1],
         breaks = breaks , show_colnames = F , treeheight_row = 0, fontsize_row = 5)
```

## Proteasome
#### KEGG Proteoasome
```{r proteasome boxplot, fig.height=5}
proteasome_genes_mean <- assay(multiomics_MAE[proteasome_genes, ,"proteomics"]) %>% colMeans(na.rm = TRUE) %>% enframe()

proteasome_CCP_P_plot <- left_join(proteasome_genes_mean, 
          colData(multiomics_MAE) %>% as_tibble() %>% select(name=patient_ID, PG ), 
          by=c("name")) %>%
  filter(!is.na(PG)) %>%
  ggplot(aes( PG, value ))+
  geom_boxplot(aes(fill=PG)) + geom_beeswarm() + 
  ggtitle("Proteasome protein abundance") + 
  pp_sra +
  ylab("Mean abundance of proteasomal proteins")+
  #stat_compare_means(label = "p.signif", method = "t.test",
  #                   ref.group = ".all.", label.y = 0.3, hide.ns = TRUE)+
  guides(fill=guide_legend(title="PG")) +
  scale_fill_manual(values = colors_CCP)
proteasome_CCP_P_plot +   theme(aspect.ratio=1) + 
    stat_compare_means(method = "anova", label.y = 0.32, hjust=0)
```

#### Heatmap all proteasome proteins
```{r}
some_proteasome <- proteasome_genes

tmp_proteasome <- wideFormat(multiomics_MAE[some_proteasome, ,"proteomics"], colDataCols = c("PG", "IGHV_mutated" )) %>% 
  as_tibble()
tmp_proteasome_mx <- tmp_proteasome %>% dplyr::select(-primary, -PG, -IGHV_mutated) %>% as.matrix()
rownames(tmp_proteasome_mx) <- tmp_proteasome$primary
colnames(tmp_proteasome_mx) <- gsub("proteomics_", "", colnames(tmp_proteasome_mx))
tmp_proteasome_anno <- tmp_proteasome[, c("PG", "IGHV_mutated")] %>% as.data.frame()
rownames(tmp_proteasome_anno) <- tmp_proteasome$primary
tmp_proteasome_anno$PG <- factor(tmp_proteasome_anno$PG,levels = c(5,1:4,6))
tmp_proteasome_anno$IGHV[tmp_proteasome_anno$IGHV_mutated==0] <- "U-CLL"
tmp_proteasome_anno$IGHV[tmp_proteasome_anno$IGHV_mutated==1] <- "M-CLL"
tmp_proteasome_anno <- tmp_proteasome_anno %>% select(-IGHV_mutated)

breaks= seq(min(tmp_proteasome_mx), max(tmp_proteasome_mx), 0.1)^2
breaks= sort(c(-breaks, breaks))
breaks <- breaks[! (breaks < min(tmp_proteasome_mx) | breaks > max(tmp_proteasome_mx) )]

ann_colors = list(
  PG=c("1"= colors_CCP[1], "2"= colors_CCP[2], "3"= colors_CCP[3], "4"= colors_CCP[4], "5"= colors_CCP[5], "6"= colors_CCP[6] ),
  IGHV=c("U-CLL"= "#0571b0", "M-CLL"= "#ca0020"))

pat_order_hclust_splice <- sapply(c(5,1:4,6), function(P){
  hc <- hclust(dist(tmp_proteasome_mx[rownames(tmp_proteasome_anno[tmp_proteasome_anno$PG==P,]), ] ))
  hc$labels[hc$order]
}) %>% unlist

PG_proteasome_proteins_pheatmap <- pheatmap(t(tmp_proteasome_mx[pat_order_hclust_splice, ]), 
         annotation_col = tmp_proteasome_anno, annotation_colors = ann_colors, scale = "row", cluster_cols = FALSE, 
         color =  inferno(length(breaks)), border_color = NA, 
         gaps_col = (which(!tmp_proteasome_anno %>% rownames_to_column() %>% arrange(PG, IGHV) %>% .$PG %>% duplicated())-1)[-1],
         breaks = breaks , show_colnames = F , treeheight_row = 0, fontsize_row = 5)
```

## PCA
```{r PCA, fig.height=5}
prot_pca <- prcomp(t( assay(multiomics_MAE[prot_few_nas , ,"proteomics"]) ))
summary(prot_pca)
prot_pca_x <- as_tibble(prot_pca$x[,1:10])
prot_pca_x$pat_ID <- colnames(assay(multiomics_MAE[prot_few_nas , ,"proteomics"]))


prot_pca_x <- left_join(prot_pca_x, 
                        wideFormat(multiomics_MAE[, ,c("SNPs","chrom_abber", "health_record_bin") ], 
                                   colDataCols = c("gender", "treatment_status", "PG")) %>% as_tibble() , 
                        by=c("pat_ID"="primary") )

#prot_pca_x <- prot_pca_x %>% replace(is.na(.), "unknown")

ggplot(prot_pca_x, aes(PC1, PC2)) +
  geom_point(aes(color=as.factor(health_record_bin_IGHV_mutated), shape=as.factor(chrom_abber_trisomy12)))  + pp_sra +
  scale_shape_manual(values = c( 16,  1, 4)) + 
  scale_color_manual(values = c( "#0571b0","#ca0020", "grey")) + theme(aspect.ratio = 1)
ggplot(prot_pca_x, aes(PC1, PC3)) +geom_point(aes(color=as.factor(health_record_bin_IGHV_mutated), shape=as.factor(chrom_abber_trisomy12)) ) +
  scale_shape_manual(values = c( 16,  1, 4)) + 
  scale_color_manual(values = c( "#0571b0","#ca0020", "grey"))+ pp_sra+ theme(aspect.ratio = 1)
ggplot(prot_pca_x, aes(PC2, PC3)) +geom_point(aes(color=as.factor(health_record_bin_IGHV_mutated), shape=as.factor(chrom_abber_trisomy12)) ) +  
  scale_shape_manual(values = c( 16,  1, 4)) + 
  scale_color_manual(values = c( "#0571b0", "#ca0020", "grey")) + pp_sra+ theme(aspect.ratio = 1)

ggplot(prot_pca_x, aes(PC1, PC2)) +geom_point(aes(color=gender)) + pp_sra + scale_color_manual(values = c( "#0571b0","#ca0020", "grey")) + theme(aspect.ratio = 1)

PCA_CCP_1_2 <- ggplot(prot_pca_x, aes(PC1, PC2)) +
  geom_point(aes(fill=PG),shape=21, color="grey") + pp_sra  +
  guides(color=guide_legend(title="PG")) +
  scale_fill_manual(values = colors_CCP) +
  xlab( paste0("PC1 (", round(summary(prot_pca)$importance[2,1], 2)*100, "%)" ))+
  ylab( paste0("PC2 (", round(summary(prot_pca)$importance[2,2], 2)*100, "%)" ))
PCA_CCP_1_2 +
  theme(aspect.ratio=1, plot.title = element_text(size = 30))

PCA_CCP_1_3 <- ggplot(prot_pca_x, aes(PC1, PC3)) +
  geom_point(aes(fill=PG),shape=21, color="grey") + pp_sra  +
  guides(color=guide_legend(title="PG")) +
  scale_fill_manual(values = colors_CCP)  +
  xlab( paste0("PC1 (", round(summary(prot_pca)$importance[2,1], 2)*100, "%)" ))+
  ylab( paste0("PC3 (", round(summary(prot_pca)$importance[2,3], 2)*100, "%)" ))
PCA_CCP_1_3 +
  theme(aspect.ratio=1, plot.title = element_text(size = 30))

PCA_CCP_2_3 <- ggplot(prot_pca_x, aes(PC2, PC3)) +
  geom_point(aes(fill=PG),shape=21, color="grey") + pp_sra  +
  guides(color=guide_legend(title="PG")) +
  scale_fill_manual(values = colors_CCP)  +
  xlab( paste0("PC2 (", round(summary(prot_pca)$importance[2,2], 2)*100, "%)" ))+
  ylab( paste0("PC3 (", round(summary(prot_pca)$importance[2,3], 2)*100, "%)" ))
PCA_CCP_2_3 +
  theme(aspect.ratio=1, plot.title = element_text(size = 30))

PCA_CCP_1_4 <- ggplot(prot_pca_x, aes(PC1, PC4)) +
  geom_point(aes(fill=PG),shape=21, color="grey") + pp_sra  +
  guides(color=guide_legend(title="PG")) +
  scale_fill_manual(values = colors_CCP)  +
  xlab( paste0("PC1 (", round(summary(prot_pca)$importance[2,1], 2)*100, "%)" ))+
  ylab( paste0("PC4 (", round(summary(prot_pca)$importance[2,4], 2)*100, "%)" ))
PCA_CCP_1_4 +
  theme(aspect.ratio=1, plot.title = element_text(size = 30))
```

## tsne RNA
```{r tsne RNA, message=FALSE, fig.height=5}
genes_no_nas <- multiomics_MAE[["RNAseq_norm"]] %>% is.na() %>% rowSums()
genes_no_nas <- genes_no_nas[ genes_no_nas == 0 ] %>% names()

set.seed(10)
rtsne_out_RNA <- Rtsne( t(assay(multiomics_MAE[genes_no_nas , ,"RNAseq_norm"])), perplexity = 10 )
rtsne_out_RNA_df <- rtsne_out_RNA$Y %>% as.data.frame() %>% as_tibble()
rtsne_out_RNA_df$pat_ID <- colnames(assay(multiomics_MAE[genes_no_nas , ,"RNAseq_norm"]))

rtsne_out_RNA_df <- left_join(rtsne_out_RNA_df, 
                          (wideFormat(multiomics_MAE[, ,c("SNPs","chrom_abber", "health_record_bin") ], colDataCols = c("gender", "treatment_status", "doehner_groups", "PG", "CCP6_RNA")) %>% as_tibble()), 
                          by=c("pat_ID"="primary") )

rtsne_out_RNA_df <- mutate_at(rtsne_out_RNA_df, colnames(rtsne_out_RNA_df %>% dplyr::select(SNPs_ATM:health_record_bin_treated)), as.logical)

#rtsne_out_RNA_df <- rtsne_out_RNA_df %>% replace(is.na(.), "unknown")

tsne_CCP_R_plot <- ggplot(rtsne_out_RNA_df, aes(V1, V2)) +
  geom_point(aes(fill=PG), color="grey", shape=21) +
  scale_fill_manual(values = colors_CCP)+
  pp_sra +
  guides(color=guide_legend(title="PG"))
tsne_CCP_R_plot + theme(aspect.ratio=1, plot.title = element_text(size = 30))

tsne_CCPRNA_R_plot <- ggplot(rtsne_out_RNA_df, aes(V1, V2)) +
  geom_point(aes(fill=CCP6_RNA), color="grey", shape=21) +
  scale_color_hue()+
  pp_sra +
  guides(color=guide_legend(title="CC_RNA"))
tsne_CCPRNA_R_plot+  theme(aspect.ratio=1, plot.title = element_text(size = 30))

sapply(colnames(rtsne_out_RNA_df)[c(4,5,9:16)], function(var){
  print(ggplot(rtsne_out_RNA_df, aes(V1, V2)) +
          geom_point(aes_string(color=var), size=3) +
    scale_color_manual(values = c("#92c5de", "#f4a582",  "grey"))  + 
          ggtitle(gsub("SNPs_", "", gsub("chrom_abber_", "", gsub( "health_record_bin_", "", var) ))) +
      pp_sra_noguides +
      theme(aspect.ratio=1, plot.title = element_text(size = 30)) )
})

```

#### Color tsne based on expression of B-cell receptor signaling pathways
```{r tsne RNA BCR genes, message=FALSE, fig.height=5, fig.height=5}
BCR_genes_RNA <- left_join(enframe(BCR_genes, value = "symbol", name = NULL), metadata(multiomics_MAE)$gene_symbol_mapping, by=c("symbol"="hgnc_symbol")) %>% 
  filter(!is.na(ensembl_gene_id)) %>% .$ensembl_gene_id
BCR_genes_RNA_mean <- assay(multiomics_MAE[BCR_genes_RNA, ,"RNAseq_norm"]) %>% colMeans(na.rm = TRUE) %>% enframe()
```

# Save important plots
```{r save image}
save(tsne_CCP_P_plot, tsne_CCP_R_plot,tsne_CCPRNA_R_plot, tsne_IGHVtris_plot, BCR_CCP_P_plot, splicing_CCP_P_plot,
     PCA_CCP_1_2, PCA_CCP_1_3, PCA_CCP_1_4, PCA_CCP_2_3, 
     PG_BCR_proteins_pheatmap, PG_BCR_proteins_pheatmap_notscaled,
     PG_splice_proteins_pheatmap, PG_splice_proteins_pheatmap_notscaled, PG_BCAA_proteins_pheatmap,
     BCAA_CCP_P_plot, proteasome_CCP_P_plot, PG_proteasome_proteins_pheatmap,
file = "RData_plots/CLL_Proteomics_DimensionReduction_Plots.RData")
```

# Session Info
```{r sessionInfo}
sessionInfo()
```
