---
title: "Validation of PG5 in Eagle et al. dataset"
author: "Sophie Rabe"
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_depth: 3
    toc_float: yes
    df_print: paged
  word_document:
    toc: yes
    toc_depth: '4'
  pdf_document:
    toc: yes
    toc_depth: '4'
editor_options:
  chunk_output_type: console
---
Analysis date: `r Sys.Date()`

# Setup
## Load libraries
```{r, message=FALSE, warning=FALSE}
library(tidyverse)
library(limma)
library(ggbeeswarm)
library(MultiAssayExperiment)
library(pheatmap)
library(ggpubr)
library(readxl)
library(biomartr)
library(biomaRt)
```

## Load data
```{r}
source("data/Figure_layouts.R")
load("data/CLL_Proteomics_Setup.RData")
load("data/CLL_Proteomics_ConsensusClustering.RData")

colData(multiomics_MAE)$PG <- as.factor(CCP_group5[rownames(colData(multiomics_MAE))])

Eagle2015 <- read_excel("data/2015Eagle_ProteinAbundances.xls")
```

## Format data
```{r, eval=FALSE}
#ensembl=useMart("ensembl")
#ensembl = useDataset("hsapiens_gene_ensembl",mart=ensembl)
#Swiss_to_hgnc <- getBM(attributes = c("hgnc_symbol", "uniprotswissprot"), 
#                 filters = "uniprotswissprot", values = Eagle2015$`SwissProt Acc. No.`, mart = ensembl)
#save(Swiss_to_hgnc, file="/Volumes/sd17b003/Sophie/Analysis/CLL_Proteomics/CLL_Proteomics_final/Robjects/Swiss_to_hgnc.RData")
```

```{r}
load("data/Swiss_to_hgnc.RData")

Eagle2015 <- left_join(Eagle2015, Swiss_to_hgnc, by=c("SwissProt Acc. No."="uniprotswissprot"))

ann_colors = list(
  PG=c("1"= colors_CCP[1], "2"= colors_CCP[2], "3"= colors_CCP[3], "4"= colors_CCP[4], "5"= colors_CCP[5], "6"= colors_CCP[6] ),
  IGHV=c("U"= "#0571b0", "M"= "#ca0020"),
  pathway = c("splicing" = "#cc99ff", "BCR"= "#66ff99" ),
  trisomy12 = c("1" = "gray40", "0" = "gray90" ),
  upper_20_perc_chr12 = c("1" = "gray40", "0" = "gray90" ),
  cluster = c("PG5 like"= colors_CCP[5], "other" = "gray") )
```

# Analysis
## Plot proteins our data
```{r}
splice_BCR_Rabe_mx <- assay(multiomics_MAE[ c(splice_genes, BCR_genes), , "proteomics" ])
dim(splice_BCR_Rabe_mx)
splice_BCR_Rabe_anno <- colData(multiomics_MAE[,  colnames(splice_BCR_Rabe_mx),] )[ , c("PG", "IGHV", "trisomy12")] 
splice_BCR_Rabe_anno$trisomy12 <- as.factor(splice_BCR_Rabe_anno$trisomy12)
splice_BCR_Rabe_anno <- splice_BCR_Rabe_anno[,c(2,3,1)]

prot_anno <- enframe(rownames(splice_BCR_Rabe_mx), value = "hgnc_symbol")
prot_anno <- prot_anno %>% 
  mutate(pathway = if_else(hgnc_symbol %in% BCR_genes, "BCR",
                           if_else(hgnc_symbol %in% splice_genes, "splicing", "NA" ) ) )
prot_anno <- prot_anno %>% dplyr::select(-name) %>% column_to_rownames("hgnc_symbol")

breaks= seq(min(splice_BCR_Rabe_mx), max(splice_BCR_Rabe_mx), 0.1)^2
breaks= sort(c(-breaks, breaks))
breaks <- breaks[! (breaks < min(splice_BCR_Rabe_mx) | breaks > max(splice_BCR_Rabe_mx) )]

gene_order_hclust <- sapply(c("BCR", "splicing" ), function(P){
  hc <- hclust(dist(splice_BCR_Rabe_mx[rownames(prot_anno)[prot_anno$pathway==P], ] ))
  hc$labels[hc$order]
}) %>% unlist

PG5_validation_ourHeatmap <- pheatmap(splice_BCR_Rabe_mx[gene_order_hclust, ], 
         annotation_col  =  as.data.frame(splice_BCR_Rabe_anno), 
         annotation_colors = ann_colors,
         annotation_row = prot_anno,
         color = viridis::inferno(length(breaks)), border_color = NA,
         breaks = breaks,
         cluster_rows = FALSE,
         cutree_cols = 2,
         show_rownames = F, 
         show_colnames = F,
         annotation_legend= F)

PG5_validation_ourHeatmap_legend <- pheatmap(splice_BCR_Rabe_mx[gene_order_hclust, ], 
         annotation_col  =  as.data.frame(splice_BCR_Rabe_anno), 
         annotation_colors = ann_colors,
         annotation_row = prot_anno,
         color = viridis::inferno(length(breaks)), border_color = NA,
         breaks = breaks,
         cluster_rows = FALSE,
         cutree_cols = 2,
         show_rownames = F, 
         show_colnames = F,
         legend= T, silent = T)
```

## Eagle et al. Splicing and BCR proteins
```{r}
# Select genes
splice_BCR_Eagle_mx <- Eagle2015 %>% 
  filter(hgnc_symbol %in% c(splice_genes, BCR_genes))
splice_BCR_Eagle_mx <- splice_BCR_Eagle_mx %>% 
  dplyr::select(-(c(N:`P Value`, `Protein Name`, `SwissProt Acc. No.` ))) %>%
  group_by(hgnc_symbol) %>%
  dplyr::summarise_all(mean) %>%
  ungroup() %>%
  column_to_rownames("hgnc_symbol") %>%
  as.matrix()

# Handle NAs
table(is.na(splice_BCR_Eagle_mx))
splice_BCR_Eagle_mx <- splice_BCR_Eagle_mx[rowSums(is.na(splice_BCR_Eagle_mx)) < 5, ]
dim(splice_BCR_Eagle_mx)
protMedian <- splice_BCR_Eagle_mx %>% rowMedians(na.rm = TRUE)
for(i in 1:nrow(splice_BCR_Eagle_mx)){
  splice_BCR_Eagle_mx[i, which(is.na(splice_BCR_Eagle_mx[i,]))] <- protMedian[i]
}
table(is.na(splice_BCR_Eagle_mx))
dim(splice_BCR_Eagle_mx)

# Annotate genes
prot_anno <- enframe(rownames(splice_BCR_Eagle_mx), value = "hgnc_symbol")
prot_anno <- prot_anno %>% 
  mutate(pathway = if_else(hgnc_symbol %in% BCR_genes, "BCR",
                           if_else(hgnc_symbol %in% splice_genes, "splicing", "NA" ) ) )
prot_anno <- prot_anno %>% dplyr::select(-name) %>% column_to_rownames("hgnc_symbol")

chr12_mean <- left_join(Eagle2015, metadata(multiomics_MAE)$gene_symbol_mapping[, c("hgnc_symbol", "chromosome_name")]) %>%
  filter(chromosome_name == 12) %>%
  gather("Pat", "Value", M1:UM9) %>%
  group_by(hgnc_symbol, Pat, chromosome_name ) %>%
  dplyr::summarise(Value = mean(Value)) %>% 
  ungroup() %>%
  group_by(Pat) %>%
  dplyr::summarise(mean_chr12 = mean(Value, na.rm=TRUE))

pat_anno <- enframe(colnames(splice_BCR_Eagle_mx), value = "Pat") %>% 
  mutate(IGHV = if_else( grepl("UM", Pat), "U", "M")) %>%
  dplyr::select(-name) %>%
  left_join(chr12_mean) %>%
  mutate(upper_20_perc_chr12 = as.factor(as.numeric(mean_chr12 >= quantile(chr12_mean$mean_chr12, probs = c(0.8) ) ) ) ) %>%
  dplyr::select(-mean_chr12) %>%
  column_to_rownames("Pat") 

breaks= seq(min(splice_BCR_Eagle_mx), max(splice_BCR_Eagle_mx), 0.1)^2
breaks= sort(c(-breaks, breaks))
breaks <- breaks[! (breaks < min(splice_BCR_Eagle_mx) | breaks > max(splice_BCR_Eagle_mx) )]

gene_order_hclust <- sapply(c("BCR", "splicing" ), function(P){
  hc <- hclust(dist(splice_BCR_Eagle_mx[rownames(prot_anno)[prot_anno$pathway==P], ] ))
  hc$labels[hc$order]
}) %>% unlist

PG5_validation_EagleHeatmap <- pheatmap(splice_BCR_Eagle_mx[gene_order_hclust,], 
         annotation_row = prot_anno, 
         annotation_col = pat_anno, 
         annotation_colors = ann_colors,
         color = viridis::inferno(length(breaks)), border_color = NA,
         breaks = breaks,
         cluster_rows = FALSE, 
         cutree_cols = 2,
         show_rownames = F, 
         show_colnames = F,
         annotation_legend= F,
         annotation_names_col = F,
         silent = T)

clusters <- cutree(PG5_validation_EagleHeatmap$tree_col, k=2) %>% enframe(name = "patient", "cluster")
clusters <- clusters %>% 
  mutate(cluster = if_else( cluster== 2, "PG5 like", "other" ))
pat_anno <- left_join( pat_anno %>% rownames_to_column("patient"), clusters) %>%
  column_to_rownames("patient")

PG5_validation_EagleHeatmap <- pheatmap(splice_BCR_Eagle_mx[gene_order_hclust,], 
         annotation_row = prot_anno, 
         annotation_col = pat_anno, 
         annotation_colors = ann_colors,
         color = viridis::inferno(length(breaks)), border_color = NA,
         breaks = breaks,
         cluster_rows = FALSE, 
         cutree_cols = 2,
         show_rownames = F, 
         show_colnames = F,
         annotation_legend= F,
         silent = F)


PG5_validation_EagleHeatmap_legend <- pheatmap(splice_BCR_Eagle_mx[gene_order_hclust,], 
         annotation_row = prot_anno, 
         annotation_col = pat_anno, 
         annotation_colors = ann_colors,
         color = viridis::inferno(length(breaks)), border_color = NA,
         breaks = breaks,
         cluster_rows = FALSE, 
         cutree_cols = 2,
         show_rownames = F, 
         show_colnames = F,
         legend= T, silent = T)
```

```{r}
# Select genes
Eagle_BCAA <- Eagle2015 %>% 
  filter(hgnc_symbol %in% BCAA_genes) %>%
  dplyr::select(-(c(N:`P Value`, `Protein Name`, `SwissProt Acc. No.` ))) %>%
  group_by(hgnc_symbol) %>%
  dplyr::summarise_all(mean) %>%
  ungroup() %>%
  gather("patient", "value", -hgnc_symbol) %>%
  group_by(patient ) %>% 
  dplyr::summarise(BCAA = mean(value, na.rm=T) ) %>%
  left_join( clusters ) %>%
  mutate(cluster = as.factor(cluster)) %>%
  ggplot(aes( cluster, BCAA, group= cluster, fill=cluster )) +
  geom_boxplot() +
  geom_beeswarm() +
  pp_sra +
  scale_fill_manual(values = c("gray",  colors_CCP[5]) )+
  theme(legend.position = 'none') +
  ylab("Mean abundance of\nBCAA proteins")
Eagle_BCAA + stat_compare_means() 

# Select genes
Eagle_proteasome <- Eagle2015 %>% 
  filter(hgnc_symbol %in% proteasome_genes) %>%
  dplyr::select(-(c(N:`P Value`, `Protein Name`, `SwissProt Acc. No.` ))) %>%
  group_by(hgnc_symbol) %>%
  dplyr::summarise_all(mean) %>%
  ungroup() %>%
  gather("patient", "value", -hgnc_symbol) %>%
  group_by(patient ) %>% 
  dplyr::summarise(proteasome = mean(value, na.rm=T) ) %>%
  left_join( clusters ) %>%
  mutate(cluster = as.factor(cluster)) %>%
  ggplot(aes( cluster, proteasome, group= cluster, fill=cluster )) +
  geom_boxplot() +
  geom_beeswarm() +
  pp_sra +
  scale_fill_manual(values = c("gray",  colors_CCP[5]) )+
  theme(legend.position = 'none') +
  ylab("Mean abundance of\nproteasomal proteins")
Eagle_proteasome + stat_compare_means() 
```

# Save important plots
```{r}
save(PG5_validation_EagleHeatmap,
     PG5_validation_EagleHeatmap_legend,
     PG5_validation_ourHeatmap,
     PG5_validation_ourHeatmap_legend,
     Eagle_BCAA,
     Eagle_proteasome,
     file = "RData_plots/CLL_Proteomics_ValidationEagle_Plots.RData")
```

# Session Info
```{r}
sessionInfo()
```